File: NCU/sophomore/CE3005/alg/classic/02_Prime/Prime.py
from math import sqrt


def primeCheck1(n) :
    for i in range(2, n) :
        if (n % i == 0) :
            return False
    
    return True


def primeCheck2(n) :
    for i in range(2, sqrt(n)) :
        if (n % i == 0) :
            return False

    return True

File: ce/ce3005/alg/classic/02_Prime/Prime.cpp
#include <iostream>
#include <cstdlib>
#include <cmath>

using namespace std;

bool primeCheck1(int);
bool primeCheck2(int);

bool primeCheck1(int n) {
    for (int i = 2; i < n; i++) {
        if (n % i == 0) {
            return false;
        } 
    }
    return true;
}

bool primeCheck2(int n) {
    for (int i = 2; i <= sqrt(n); i++) {
        if (n % i == 0) {
            return false;
        } 
    }
    return true; 
}